File: util/java/src/main/java/com/dmi/util/io/InputStreamWrapper.java
package com.dmi.util.io;

import java.io.IOException;
import java.io.InputStream;

public abstract class InputStreamWrapper extends InputStream {
    private final InputStream stream;

    public InputStreamWrapper(InputStream stream) {
        this.stream = stream;
    }

    @Override
    public int available() throws IOException {
        return stream.available();
    }

    @Override
    public void close() throws IOException {
        stream.close();
    }

    @Override
    public void mark(int readlimit) {
        stream.mark(readlimit);
    }

    @Override
    public boolean markSupported() {
        return stream.markSupported();
    }

    @Override
    public int read(byte[] buffer) throws IOException {
        return stream.read(buffer);
    }

    @Override
    public int read(byte[] buffer, int byteOffset, int byteCount) throws IOException {
        return stream.read(buffer, byteOffset, byteCount);
    }

    @Override
    public synchronized void reset() throws IOException {
        stream.reset();
    }

    @Override
    public long skip(long byteCount) throws IOException {
        return stream.skip(byteCount);
    }

    @Override
    public int read() throws IOException {
        return stream.read();
    }
}

File: util/util-common/src/main/kotlin/com/dmi/util/io/InputStreamWrapper.kt
package com.dmi.util.io

import java.io.IOException
import java.io.InputStream

abstract class InputStreamWrapper(private val stream: InputStream) : InputStream() {

    @Throws(IOException::class)
    override fun available(): Int {
        return stream.available()
    }

    @Throws(IOException::class)
    override fun close() {
        stream.close()
    }

    override fun mark(readlimit: Int) {
        stream.mark(readlimit)
    }

    override fun markSupported(): Boolean {
        return stream.markSupported()
    }

    @Throws(IOException::class)
    override fun read(buffer: ByteArray): Int {
        return stream.read(buffer)
    }

    @Throws(IOException::class)
    override fun read(buffer: ByteArray, byteOffset: Int, byteCount: Int): Int {
        return stream.read(buffer, byteOffset, byteCount)
    }

    @Synchronized @Throws(IOException::class)
    override fun reset() {
        stream.reset()
    }

    @Throws(IOException::class)
    override fun skip(byteCount: Long): Long {
        return stream.skip(byteCount)
    }

    @Throws(IOException::class)
    override fun read(): Int {
        return stream.read()
    }
}
